src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -447,6 +442,15 @@ bool SotaUptaneClient::getNewTargets(std::vector<Uptane::Target> *new_targets, u
         is_new = true;
       }
 
+      if (primary_ecu_serial == ecu_serial) {
+        if (!target.IsOstree() &&
+            (config.pacman.type == PackageManager::kOstree || config.pacman.type == PackageManager::kOstreeDockerApp)) {
+          LOG_ERROR << "Cannot install a non-OSTree package on an OSTree system";
+          sendEvent<event::InstallTargetComplete>(primary_ecu_serial, false);
```

Comment:
IMHO, I'd make this code more generic and extendable, i.e. instead of comparing a specific target type with specific package manager types do something like the following:
`if (target.type != config.pacman.type)`

Reply:
> IMHO, I'd make this code more generic and extendable

I like the idea but currently we don't really have a good way to support that. The only type that we can recognize in the Target metadata is OSTree, so for now, that's the only one worth checking.

> In other cases when `is_new = false` the event `event::InstallTargetComplete` is not sent, I am just wondering if it's correct place to do it here.

I agree, instead of sending that event, we should just set `last_exception` and return false. The log message is fine, though.